tests/tracking/test_fluent.py

```diff
@@ -0,0 +1,71 @@
+import os
+import random
+
+import mlflow
+from mlflow.entities import Experiment
+from mlflow.tracking.fluent import _get_experiment_id, _get_experiment_id_from_env, \
+    _EXPERIMENT_NAME_ENV_VAR, _EXPERIMENT_ID_ENV_VAR
+from mlflow.utils.file_utils import TempDir
+
+
+class HelperEnv:
+    @classmethod
+    def assert_values(cls, exp_id, name):
+        assert os.environ.get(_EXPERIMENT_NAME_ENV_VAR) == name
+        assert os.environ.get(_EXPERIMENT_ID_ENV_VAR) == exp_id
+
+    @classmethod
+    def set_values(cls, id=None, name=None):
+        if id:
+            os.environ[_EXPERIMENT_ID_ENV_VAR] = str(id)
+        elif os.environ.get(_EXPERIMENT_ID_ENV_VAR):
+            del os.environ[_EXPERIMENT_ID_ENV_VAR]
+
+        if name:
+            os.environ[_EXPERIMENT_NAME_ENV_VAR] = str(name)
+        elif os.environ.get(_EXPERIMENT_NAME_ENV_VAR):
+            del os.environ[_EXPERIMENT_NAME_ENV_VAR]
+
+
+def test_get_experiment_id_from_env():
+    # When no env variables are set
+    HelperEnv.assert_values(None, None)
+    assert _get_experiment_id_from_env() is None
+
+    # set only ID
+    random_id = random.randint(1, 1e6)
+    HelperEnv.set_values(id=random_id)
+    HelperEnv.assert_values(str(random_id), None)
+    assert _get_experiment_id_from_env() == str(random_id)
+
+    # set only name
+    with TempDir(chdr=True):
```

Comment:
Curious why this line is necessary. If it's to isolate all of the different test cases from each other, it'd be awesome to have something more explicit in intent which we can use for all test cases.

Reply:
Because this part of the test is creating a new experiment. I could mock store to create and return experiments or let it create.